client/python/lib/test/test_tensors.py

```diff
@@ -75,32 +75,258 @@ def test_make_ndarray_valid_no_content_string_with_other_type_content():
+    tensor_proto = make_tensor_proto(values = array, dtype = DataType.DT_FLOAT)
+    assert tensor_proto.tensor_content == np_values.tobytes()
+    assert tensor_proto.dtype == DataType.DT_FLOAT
+    assert tensor_proto.tensor_shape == _shape
+
+def test_make_tensor_proto_valid_no_dtype_provided():
+    values = [1.0, 2.0, 3.0]
+    tensor_proto = make_tensor_proto(shape=[3], values = values)
+    np_values = np.array(values)
+    assert tensor_proto.tensor_content == np_values.tobytes()
+    assert tensor_proto.dtype == DataType.DT_DOUBLE
+    assert tensor_proto.tensor_shape == TensorShapeProto(dim = [TensorShapeProto.Dim(size=3)])
+
+def test_make_tensor_proto_valid_no_shape_provided():
+    values = [[1.0, 2.0, 3.0], [1.0, 2.0, 3.0]]
+    tensor_proto = make_tensor_proto(dtype = DataType.DT_FLOAT, values = values)
+    expected_shape = TensorShapeProto(dim = [TensorShapeProto.Dim(size=2), TensorShapeProto.Dim(size=3)])
+    np_values = np.array(values, np.float32)
+    assert tensor_proto.tensor_content == np_values.tobytes()
+    assert tensor_proto.tensor_shape == expected_shape
+    assert tensor_proto.dtype == DataType.DT_FLOAT
+
+# def test_make_tensor_proto_invalid_dtype_provided():
+#     values = [[1.0, 2.0, 3.0], [1.0, 2.0, 3.0]]
+#     with pytest.raises(TypeError) as exception_info:
+#         make_tensor_proto(shape = None, dtype = DataType.DT_STRING, values = values)
+#         exception = exception_info.value
+#         assert str(exception) == "[1.0, 2.0, 3.0] has type list, but expected one of: bytes"
+
+def test_make_tensor_proto_invalid_dimsions():
+    values = [[1.0, 2.0], [1.0, 2.0, 3.0]]
+    with pytest.raises(ValueError) as exception_info:
+        make_tensor_proto(shape=[2,3], dtype = DataType.DT_FLOAT, values = values)
+        exception = exception_info.value
+        assert str(exception) == "setting an array element with a sequence"
+
+def test_make_tensor_proto_invalid_dimsions_2():
+    values = [[(1,2,3)],[(1,2)],[(1,2,3)]]
+    with pytest.raises(ValueError) as exception_info:
+        make_tensor_proto(shape=[2,3], dtype = DataType.DT_FLOAT, values = values)
+        exception = exception_info.value
+        assert str(exception) == "setting an array element with a sequence"
+
+def test_make_tensor_proto_invalid_binary():
+    values = [1,2,3]
+    with pytest.raises(TypeError) as exception_info:
+        make_tensor_proto(shape=None, dtype = DataType.DT_STRING, values = values)
+        exception = exception_info.value
+        assert str(exception) == "1 has type int, but expected one of: bytes"
+
+def test_make_tensor_proto_invalid_shape_type():
+    values = 5.0
+    with pytest.raises(TypeError) as exception_info:
+        make_tensor_proto(shape=0, dtype = DataType.DT_FLOAT, values = values)
+        exception = exception_info.value
+        assert str(exception) == "shape type should be list or tuple, but is int"
+
+def test_make_tensor_proto_invalid_dimensions_no_shape_provided():
```

Comment:
The situation here is very similar to `test_make_tensor_proto_invalid_dimsions`. Why the error is different?